### PR TITLE
feat: achievements plugin.

### PR DESCRIPTION
**Pull request by yashdesu**
#### Checklist

**I agree to the following :-**

- [ ] Added description of the change
- [ ] I've read the [contributing guidelines](https://github.com/WebXDAO/devprotocol.xyz/blob/main/CONTRIBUTING.md)
- [ ] Search previous suggestions before making a new PR, as yours may be a duplicate.
- [ ] I acknowledge that all my contributions will be made under the project's license.

Notes (**optional**):

### CODE REVIEW

File: src/plugins/achievements/redis-schema.ts [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+ */
+export const achievementDocument = (
+  base: Omit<Achievement, 'id'>,
+): AchievementForDb => {
+  const achievement = {
+    ...base,
+    metadata: {
+      ...base.metadata,
+      numberAttributes: encode(base.metadata.numberAttributes),
+      stringAttributes: encode(base.metadata.stringAttributes),
```
Comment (aggre) [inline src/plugins/achievements/redis-schema.ts:171]: Redis expected to be able to parse JSON for TEXT index, so this lines can be the following: https://redis.io/docs/interact/search-and-query/indexing/#index-json-arrays-as-text
```ts
      numberAttributes: base.metadata.numberAttributes,
      stringAttributes: base.metadata.stringAttributes,
```

File: src/plugins/achievements/redis-schema.ts [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+
+export const metadataNumberAttributes = {
+  '$.metadata.numberAttributes': {
+    type: SchemaFieldTypes.TEXT,
+    AS: 'metadataNumberAttributes',
+  },
+} satisfies RediSearchSchema
+
+export const metadataStringAttributes = {
+  '$.membership.stringAttributes': {
```
Comment (aggre) [inline src/plugins/achievements/redis-schema.ts:106]: Is `$.membership.stringAttributes` be `$.meta.stringAttributes`?

File: src/plugins/achievements/redis-schema.ts (hunk 1 above)
Comment (yashdesu) [reply src/plugins/achievements/redis-schema.ts]: Fixed it here: https://github.com/dev-protocol/clubsx/pull/2129/files#diff-e4bbcb5ec5cea42de6f80582d340b07dce14e7496e9bb04a00d164c7fe5cf55dR163-R165)

File: src/plugins/achievements/redis-schema.ts (hunk 2 above)
Comment (yashdesu) [reply src/plugins/achievements/redis-schema.ts]: Fixed it here and also replaced type from SchemaFieldTypes.TAG to SchemaFieldTypes.TEXT https://github.com/dev-protocol/clubsx/pull/2129/files#diff-e4bbcb5ec5cea42de6f80582d340b07dce14e7496e9bb04a00d164c7fe5cf55dR106-R107

**Pull request closed without merging**